### ISSUE TITLE
Unable to ignore a path containing square brackets

### ISSUE THREAD

**Issue by robatwilliams** [kaelzhang/node-ignore#59]
I'm using the ESLint [no-restricted-imports](https://eslint.org/docs/rules/no-restricted-imports) rule, which uses this package under the hood. I've reproduced this issue against node-ignore directly.

We have some tokens in our import paths which are denoted by square brackets. Ignoring these doesn't currently seem to work. I've escaped them in the pattern as required by gitignore (and another `\` to escape the `\` for JavaScript), and tried the same scenario using `git check-ignore` which works.

```javascript
it('ignores patterns containing square brackets', () => {
  const ignorer = ignore().add(['src/\\[app\\]']);

  expect(ignorer.ignores('src/[app]')).toBe(true); // fails
});
```

Such paths also can't be un-ignored:

```javascript
it('un-ignores negated patterns containing square brackets', () => {
  const ignorer = ignore().add(['src/**', '!src/\\[app\\]']);

  expect(ignorer.ignores('src/a')).toBe(true);
  expect(ignorer.ignores('src/[app]')).toBe(false); // fails
});
```

Version: 5.1.4

**Comment by kaelzhang** [comment]
Nice catch buddy. But it will take time, I need a complete test with more test cases about escaping.

gitignore manual is really lame and lacks of cases, so I need to be cautious.

**Comment by kaelzhang** [comment]
Landed in [`5.1.5`](https://github.com/kaelzhang/node-ignore/releases/tag/5.1.5)

**Comment by kaelzhang** [comment]
However, Eslint depends on `ignore@4.x`, so the fixe is blocked by eslint/eslint#13294
